### PR TITLE
Update CI to reduce number of tests

### PR DESCRIPTION
**Pull request by blisc**
As discussed, we want to reduce our current CI load. This PR:

- separate tests into L0 and L1
- L0 should be run on all branches
- L1 should be run on master and on PRs

~Up for debate, I added a check for both unittests (under L0) and all L1 tests such that they are only run if there is a changed python file. I'm not sure if we want this. Ie if we ever add a documentation unittest.~ This feature has been removed. Tests are run regardless of files changes

### CODE REVIEW

**Comment by blisc** [comment]
If this works as expected inside Jenkins, we should see three CI tests. Two for the c_update_ci branch, and one for the PR

[x] The first run should only run PyTorch Version, Install test requirements, and Code Formatting Checks
[] The second run should additionally check unittests because I updated a python file
[] The third run should check all tests.

File: Jenkinsfile [hunk 1]
```diff
@@ -11,28 +11,38 @@ pipeline {
...
-    stage('Code formatting checks') {
+    stage('L0: Code formatting checks') {
       steps {
         sh 'python setup.py style'
       }
     }
-    stage('Unittests ALL') {
+    stage('L0: Unittests ALL') {
+      when {
+        changeset "**/*.py"
```
Comment (yzhang123) [inline Jenkinsfile]: what does when {..} do?

File: Jenkinsfile [hunk 2]
```diff
@@ -11,28 +11,38 @@ pipeline {
...
         sh './reinstall.sh && python -m unittest'
       }
     }
 
-    stage('Parallel Stage1') {
+    stage('L1: Parallel Stage1') {
+      when {
+        changeset "**/*.py"
+        anyOf{
+          branch 'master'
```
Comment (yzhang123) [inline Jenkinsfile]: what does anyof do?

File: Jenkinsfile (hunk 1 above)
Comment (okuchaiev) [reply Jenkinsfile]: @yzhang123 see https://jenkins.io/doc/book/pipeline/syntax/

File: Jenkinsfile (hunk 2 above)
Comment (okuchaiev) [reply Jenkinsfile]: https://jenkins.io/doc/book/pipeline/syntax/

**Pull request merged**